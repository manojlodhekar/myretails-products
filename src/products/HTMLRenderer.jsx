import React from 'react';

const HTMLRenderer = (props) =>{
    return(
        <div>
            <div style={{paddingTop: 10}}>
                <span id="htmlRendererId" className="returnPolicyCls" dangerouslySetInnerHTML={{__html: props.htmlContents}} />
            </div>
        </div>
    )
}
export default HTMLRenderer;