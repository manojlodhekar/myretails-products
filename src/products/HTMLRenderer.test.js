import React from 'react';
import ReactDOM from 'react-dom';
import HTMLRenderer from './HTMLRenderer';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, mount } from 'enzyme';

jest.mock('react-dom');

Enzyme.configure({ adapter: new Adapter() });
let htmlRenderer;
const htmlContent = "<a href=\"http:\/\/www.target.com\/HelpContent?help=\/sites\/html\/TargetOnline\/help\/returns_and_refunds\/returns_and_refunds.html\">Target return policy<\/a> for complete information.<\/p><br\/>";

beforeEach(()=>{
  htmlRenderer = mount(<HTMLRenderer htmlContents={htmlContent}>
  </HTMLRenderer>);

})
describe('HTMLRenderer is defined', () => {

  it('should render correctly', () => {
    expect(htmlRenderer).toHaveLength(1);
  });

  it('should render inner component', () => {
    expect(htmlRenderer).toMatchSnapshot();
  });
  it('should render html of component', () => {
    htmlRenderer = shallow(<HTMLRenderer  htmlContents={htmlContent} />);
    expect(htmlRenderer.find('.returnPolicyCls')).toHaveLength(1);
    //expect(htmlRenderer.find('#htmlRendererId').text()).toEqual(htmlContent);
  });
 });
