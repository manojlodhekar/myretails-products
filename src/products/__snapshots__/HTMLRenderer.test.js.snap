// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HTMLRenderer is defined should render inner component 1`] = `
ReactWrapper {
  "length": 1,
  Symbol(enzyme.__unrendered__): <HTMLRenderer
    htmlContents="<a href=\\"http://www.target.com/HelpContent?help=/sites/html/TargetOnline/help/returns_and_refunds/returns_and_refunds.html\\">Target return policy</a> for complete information.</p><br/>"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
